### ISSUE TITLE
Using docker-java lib run in container and get nothing from unix socket

### ISSUE THREAD

**Issue by qyvlik** [docker-java/docker-java#1190]
I have a java program which use the docker-java lib, the java program provider the docker-java lib ability pass by websocket. 
If I debug in IDEA, I can get the docker logs. Then I run the java program in docker, and get nothing from unix socket.

docker-compose.yml as follow, and I'm sure the java have the permissions to call the `/var/run/docker.sock`. Any Suggestions?

```
version: '3'

services:
  logs-viewer:
    privileged: true
    image: qyvlik/docker-simple-logs-viewer:latest
    ports:
      - "19102:19102"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./application.yml:/home/www/application.yml:ro
``` 

The qyvlik/docker-simple-logs-viewer:latest source code in here: https://github.com/qyvlik/docker-simple-logs-viewer

docker-java version 

```
        <dependency>
            <groupId>com.github.docker-java</groupId>
            <artifactId>docker-java</artifactId>
            <!-- use latest version https://github.com/docker-java/docker-java/releases -->
            <version>3.1.1</version>
        </dependency>
```

**Comment by qyvlik** [comment]
If I use the `jnr-unixsocket` to call docker api, it can return data. Code as follow

```
    @PostConstruct
    public void tryGetThingFormUnixSocket() throws IOException {
        File sockFile = new File("/var/run/docker.sock");
        UnixSocketAddress address = new UnixSocketAddress(sockFile);
        UnixSocketChannel channel = UnixSocketChannel.open(address);
        UnixSocket unixSocket = new UnixSocket(channel);

        // call Docker API
        PrintWriter w = new PrintWriter(unixSocket.getOutputStream());
        w.println("GET /v1.24/containers/json HTTP/1.1");
        w.println("Host: http");
        w.println("Accept: */*");
        w.println("");
        w.flush();
        unixSocket.shutdownOutput();

        System.out.println("---- Docker Response ----");
        BufferedReader br = new BufferedReader(new InputStreamReader(unixSocket.getInputStream()));
        String line;
        while ((line = br.readLine()) != null){
            System.out.println(line);
        }
        unixSocket.close();
    }
```

maven

```xml
        <dependency>
            <groupId>com.github.jnr</groupId>
            <artifactId>jnr-unixsocket</artifactId>
            <version>0.18</version>
        </dependency>
```

**Comment by qyvlik** [comment]
I fix the docker-java version to `3.1.2`, and java program work now.

```
        <dependency>
            <groupId>com.github.docker-java</groupId>
            <artifactId>docker-java</artifactId>
            <!-- use latest version https://github.com/docker-java/docker-java/releases -->
            <version>3.1.2</version>
        </dependency>
```
